fds/services/dvc_service.py

```diff
@@ -65,85 +74,103 @@ def __skip_already_added(self, root, dirs) -> None:
             if self.__should_skip_list_add(dir):
                 dirs.remove(d)
 
-    def __get_to_add_to_dvc(self, file_or_dir_to_check: str, dirs: List[str], type: str) -> Optional[str]:
+    @staticmethod
+    def _get_choice(file_or_dir_to_check: str, path_size: int, type: str) -> dict:
+        choices = [{
+            "key": "d",
+            "name": "Add to DVC",
+            "value": DvcChoices.ADD_TO_DVC.value
+        },{
+            "key": "g",
+            "name": "Add to Git",
+            "value": DvcChoices.ADD_TO_GIT.value
+        },{
+            "key": "i",
+            "name": "Ignore - Add to .gitignore",
+            "value": DvcChoices.IGNORE.value
+        }]
+        if os.path.isdir(file_or_dir_to_check):
+            choices.append({
+                "key": "s",
+                "name": "Step Into",
+                "value": DvcChoices.STEP_INTO.value
+            })
+
+        questions = [
+            {
+                "type": "expand",
+                "message": f"What would you like to do with {type} {file_or_dir_to_check} of {convert_bytes_to_readable(path_size)}?",
+                "name": "selection_choice",
+                "choices": choices,
+                "default": DvcChoices.ADD_TO_DVC.value
+            }
+        ]
+        answers = PyInquirer.prompt(questions)
+        return answers
+
+    def __get_to_add_to_dvc(self, file_or_dir_to_check: str, dirs: List[str], type: str, skipped_dirs: List[str]) -> Optional[str]:
         if not self.__should_skip_list_add(file_or_dir_to_check):
-            dir_size = get_size_of_path(file_or_dir_to_check)
-            if dir_size < MAX_THRESHOLD_SIZE:
+            path_size = get_size_of_path(file_or_dir_to_check)
+            # Dont need to traverse deep in case of dir
+            if path_size < MAX_THRESHOLD_SIZE:
```

Comment:
No, this is correct, this is meant to improve performance where if a directory is less than the threshold size, we still iterate on every file inside it, to prevent that I'm adding some more conditions

Reply:
Ah OK I though that the `path_size` refers to the length of the path, and the comment above is what confused me